src/rest.ts

```diff
@@ -148,6 +148,38 @@ ${queryParams
   }
 }
 
+function createScopedRequestHandler(
+  this: unknown,
+  method: RESTMethods,
+  rootURL: string,
+) {
+  const createdHandler = createRestHandler(method)
+
+  return <RequestBodyType = DefaultRequestBodyType, ResponseBodyType = any>(
+    path: string,
+    resolver: ResponseResolver<
+      MockedRequest<RequestBodyType>,
+      typeof restContext,
+      ResponseBodyType
+    >,
+  ): RequestHandler<
+    MockedRequest<RequestBodyType>,
+    typeof restContext,
+    ParsedRestRequest,
+    MockedRequest<RequestBodyType>,
+    ResponseBodyType
+  > => {
+    if (typeof rootURL !== 'string' || typeof path !== 'string') {
+      throw new Error('When using rest.link all paths must be strings.')
```

Comment:
I'm not sure here. Maybe we can do 2 pass match when using regexp expressions?  Or maybe we can do `Mask` for root path and then string only for the added bit?